### PR TITLE
Use multivariate polynomial rings throughout

### PR DESCRIPTION
**Pull request by yuan-zhou**
Call the `PolynomialRing` factory function in a way that ensures multivariate polynomial rings are created even when there is only a single variable. This is a workaround for the inconsistent/incomplete API of univariate polynomial rings in Sage.

Some doctests are adjusted to account for the different sorting order of elements after this change.

Also, a py3 fix for use of lrslib.

### CODE REVIEW

File: cutgeneratingfunctionology/spam/parametric_real_field_element.py [hunk 1]
```diff
@@ -56,9 +56,9 @@ def __init__(self, parent, value, symbolic=None):
             ## Test coercing the value to RR, so that we do not try to build a ParametricRealFieldElement
             ## from something like a tuple or vector or list or variable of a polynomial ring
             ## or something else that does not make any sense.
-            from sage.interfaces.mathematica import MathematicaElement
-            if not isinstance(value, MathematicaElement):
-                RR(value)
+            # FIXME: parent(value) caused SIGSEGV because of an infinite recursion
+            if not ((type(value) is int) or hasattr(value, 'parent') and RR.has_coerce_map_from(value.parent())):
```
Comment (mkoeppe) [inline cutgeneratingfunctionology/spam/parametric_real_field_element.py:60]: `type(value) is int`?? I think this needs a more general solution

Comment (mkoeppe) [reply cutgeneratingfunctionology/spam/parametric_real_field_element.py:60]:
```
sage: from sage.structure.coerce import py_scalar_parent                                                                                                                                                                                       
sage: py_scalar_parent(type(1r))                                                                                                                                                                                                               
Integer Ring
sage: py_scalar_parent(type([1, 2]))                                                                                                                                                                                                           
```

Comment (yuan-zhou) [reply cutgeneratingfunctionology/spam/parametric_real_field_element.py:60]: Doesn't work
sage: value = 1                                                                                                 
sage: type(value)                                                                                               
<class 'sage.rings.integer.Integer'>
sage: py_scalar_parent(type(value))

Comment (mkoeppe) [reply cutgeneratingfunctionology/spam/parametric_real_field_element.py:60]: value is already an element. this is for values that are not yet

**Pull request merged**